### PR TITLE
Logging refactor

### LINKED ISSUE #109: Logging refactor
We should set up logging more cleanly, as done in EdinburghGenomics/Reporting-App. All log files should have headers, and modules like `clarity` and `rest_communication` should not log errors during, e.g, `analysis_driver --report`.

### PR DESCRIPTION
**Pull request by mwhamgenomics**
Closes #109

### CODE REVIEW

File: analysis_driver/dataset_scanner.py [hunk 1]
```diff
@@ -71,29 +38,28 @@ def dataset_status(self):
...
-        # to an appropriate endpoint
+        # proc_id is now different, so a new analysis_driver_proc will be created
+        ntf.start_pipeline()
         self._change_status(DATASET_PROCESSING, finish=False)
 
-    def succeed(self):
+    def succeed(self, quiet=False):
         assert self.dataset_status == DATASET_PROCESSING
         self._change_status(DATASET_PROCESSED_SUCCESS)
+        if not quiet:
```
Comment (tcezard) [inline analysis_driver/dataset_scanner.py]: What is the idea behind the quiet options?

File: analysis_driver/app_logging.py [hunk 2]
```diff
@@ -1,57 +1,109 @@
...
+            datefmt=cfg.query('logging', 'datefmt', ret_default='%Y-%b-%d %H:%M:%S')
+        )
+        self.blank_formatter = logging.Formatter()
+        self.formatter = self.default_formatter
+        self.handlers = set()
+        self.loggers = set()
+        self.log_level = logging.INFO
+
+    def get_logger(self, name, level=None):
+        """
```
Comment (mwhamgenomics) [inline analysis_driver/app_logging.py:21]: Would we ever need to set up a logger without registering it to all the handlers? (e.g. clarity, rest_communication)

File: analysis_driver/notification/notification_center.py [hunk 3]
```diff
@@ -60,16 +59,11 @@ def start_stage(self, stage_name):
     def end_stage(self, stage_name, exit_status=0):
         pass
```
Comment (mwhamgenomics) [inline analysis_driver/notification/notification_center.py:61]: We should remove `self.dataset` from Notification

File: analysis_driver/quality_control/gender_validation.py [hunk 4]
```diff
@@ -64,19 +63,23 @@ def join(self, timeout=None):
         super().join(timeout=timeout)
         if self.exception:
             raise self.exception
-        return self.return_value
+        return self.exit_status
 
 
 def main():
```
Comment (mwhamgenomics) [inline analysis_driver/quality_control/gender_validation.py:69]: This should be moved to run_qc.py (not in this branch, though)

**Pull request merged**